src/core/Form/FilterInput/FilterInput.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { Component, ChangeEvent } from 'react';
+import { default as styled } from 'styled-components';
+import { withSuomifiDefaultProps } from '../../theme/utils';
+import {
+  HtmlInput,
+  HtmlInputProps,
+  HtmlDiv,
+  HtmlDivProps,
+} from '../../../reset';
+import { TokensProp, InternalTokensProp } from '../../theme';
+import { baseStyles } from './FilterInput.baseStyles';
+import classnames from 'classnames';
+import { Omit } from '../../../utils/typescript';
+import { LabelText, LabelMode } from '../LabelText/LabelText';
+import { AutoId } from '../../../utils/AutoId';
+import { InputStatus } from '../types';
+import { StatusText } from '../StatusText/StatusText';
+import { getConditionalAriaProp } from '../../../utils/aria';
+
+const baseClassName = 'fi-filter-input';
+const filterInputClassNames = {
+  error: `${baseClassName}--error`,
+  disabled: `${baseClassName}--disabled`,
+  labelAlignLeft: `${baseClassName}--label-align-left`,
+  wrapper: `${baseClassName}_wrapper`,
+  functionalityContainer: `${baseClassName}_functionalityContainer`,
+  inputElementContainer: `${baseClassName}_input-element-container`,
+  inputElement: `${baseClassName}_input`,
+};
+
+type FilterInputStatus = Exclude<InputStatus, 'success'>;
```

Comment:
Is Exclude use intentionally or should Omit be used here as well?

Reply:
I do not think it works correctly with `Omit`.

- [Omit takes an object type and removes the given properties from the type.](https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys)
- [Exclude takes a union type and removes given items of that union.](https://www.typescriptlang.org/docs/handbook/utility-types.html#excludetype-excludedunion)